pkgs/racket-doc/scribblings/reference/equality.scrbl

```diff
@@ -129,16 +128,20 @@ The behavior of a datatype with respect to @racket[eq?] is generally
 specified with the datatype and its associated procedures.
 
 
-@section{Equality-Based Hashing}
+@section{Equality and Hashing}
 
 
 All comparable values have at least one @deftech{hash code} --- an arbitrary
 integer (more specifically a @tech{fixnum}) computed by applying a hash function
-to the value. Hash codes can be used to speed up comparison tests because @bold{
- equal values have equal hash codes}, though the reverse is not necessarily
-true. The different comparison procedures each come with different functions for
-extracting hash codes. Hash codes are also used internally by @tech{hash
- tables}, see @secref["hashtables"] for more information.
+to the value. Hash codes are used internally by the various equality functions
+as an optimistic shortcut, because @bold{equal values have equal hash codes}.
```

Comment:
Hash code are only useful as a shortcut if they're stored alongside a value, as they might be in a hash table. Otherwise, the work to compute a hash code is the same kind of work as for checking equality, so it doesn't help as a shortcut.

Reply:
Huh, I wonder how I got that notion into my head. Removed the text about hash codes being used as part of checking equality. Since the only use for hash codes left to talk about is hash tables, I merged this back into a single paragraph.